libmachine/drivers/drivers.go

```diff
@@ -120,10 +120,27 @@ func DriverUserdataFlag(d Driver) string {
 	return ""
 }
 
+// DriverOSFlag returns true if the driver is detected to have an OS flag.
+func DriverOSFlag(d Driver) string {
+	for _, opt := range d.GetCreateFlags() {
+		if nameIsOS(opt.String()) {
+			return opt.String()
+		}
+	}
+
+	return ""
+}
+
 // nameIsUserData returns true if the given flag is a userdata flag
 func nameIsUserData(name string) bool {
 	return strings.Contains(name, "user-data") ||
 		strings.Contains(name, "userdata") ||
 		strings.Contains(name, "custom-data") ||
 		strings.Contains(name, "cloud-config")
 }
+
+// nameIsUserData returns true if the given flag is a userdata flag
+func nameIsOS(name string) bool {
+	return strings.Contains(name, "os") ||
```

Comment:
`strings.Contains(name, "os")` here is way too vague. This will match things like `post`, `ghost`, `most`...

Reply:
I have addressed this with `strings.HasSuffix(name, "-os")`